src/App.jsx

```diff
@@ -6,22 +6,53 @@ import './App.css'
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
+function App() {
 
-const useSemiPersistentState = () => {
-  const [todoList,setTodoList] = useState(JSON.parse(localStorage.getItem('savedTodoList')) || []);
+  const [todoList,setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
 
-  useEffect(() => {
-    localStorage.setItem('savedTodoList', JSON.stringify(todoList))
-  }, [todoList])
 
-  return [todoList, setTodoList]
+  const fetchData = async() => {
+    const options = {
+      method: "GET",
+      headers: {
+        Authorization:`Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`
+      }
+    }
+    let url = `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}`
 
-}
+    try {
+      const response = await fetch(url, options)
 
-function App() {
-  const [count, setCount] = useState(0)
+      if (!response.ok) {
+        throw new Error(`An error occurred: ${response.status}`)
+      }
+
+      const data = await response.json()
+
+      const todos = data.records.map((todo) => {
+        return {id: todo.id, title: todo.fields.Title}
```

Comment:
small issue: assignment tell to 
<img width="579" alt="image" src="https://github.com/user-attachments/assets/4dfe7728-d95c-4e28-b811-881b428d62bd">
You names the column as a Title) It is not a problem at all